src/Controllers/User.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Controllers;
+use App\Models\Main;
+use App\Models\Db;
+
+class User extends Main
+{
+
+    protected static $table='users';
```

Comment:
Attention tu es POO il manque donc dans tes fonctions le type de tes paramètres et le type de retour de tes fonctions.

```suggestion
    protected static string $table='users';
```

Reply:
ca ne devrait pas être dans la classe USER. Les notions de base de données doivent être dans une classe à part. Par exemple un UserRepository dont le seul objectif de cette classe serait d'utiliser une source de données pour récupérer les informations des users ou de mettre à jour la source de données (ici la base de données).

https://www.mitrais.com/news-updates/repository-pattern-implementation-in-php/

Reply:
Je sais mais à ce stade je réfléchis pas, je suis juste la leçon, où le gars simule ce qui est mal fait pour le refaire mieux ensuite, ce qui est assez lassant d'ailleurs. En l'occurrence cette classe est l'objet retourné par Pdo, il est donc absurde d'y loger des fonctions d'appel à Pdo...

Reply:
et pourquoi pas plutôt mettre dans un dossier spécifique les classes vont utiliser Pdo, qui sont appellées par le getter, et laisser ça là, dans son dossier "controller", puisque c'est ce que ça fait ?